packages/flutter/lib/src/cupertino/app.dart

```diff
@@ -0,0 +1,358 @@
+  /// [Navigator.push] and [Navigator.pop], will not function.
+  final TransitionBuilder builder;
+
+  /// {@macro flutter.widgets.widgetsApp.title}
+  ///
+  /// This value is passed unmodified to [WidgetsApp.title].
+  final String title;
+
+  /// {@macro flutter.widgets.widgetsApp.onGenerateTitle}
+  ///
+  /// This value is passed unmodified to [WidgetsApp.onGenerateTitle].
+  final GenerateAppTitle onGenerateTitle;
+
+  /// {@macro flutter.widgets.widgetsApp.color}
+  final Color color;
+
+  /// {@macro flutter.widgets.widgetsApp.locale}
+  final Locale locale;
+
+  /// {@macro flutter.widgets.widgetsApp.localizationsDelegates}
+  final Iterable<LocalizationsDelegate<dynamic>> localizationsDelegates;
+
+  /// {@macro flutter.widgets.widgetsApp.localeResolutionCallback}
+  ///
+  /// This callback is passed along to the [WidgetsApp] built by this widget.
+  final LocaleResolutionCallback localeResolutionCallback;
+
+  /// {@macro flutter.widgets.widgetsApp.supportedLocales}
+  ///
+  /// It is passed along unmodified to the [WidgetsApp] built by this widget.
+  final Iterable<Locale> supportedLocales;
+
+  /// Turns on a performance overlay.
+  ///
+  /// See also:
+  ///
+  ///  * <https://flutter.io/debugging/#performanceoverlay>
+  final bool showPerformanceOverlay;
+
+  /// Turns on checkerboarding of raster cache images.
+  final bool checkerboardRasterCacheImages;
+
+  /// Turns on checkerboarding of layers rendered to offscreen bitmaps.
+  final bool checkerboardOffscreenLayers;
+
+  /// Turns on an overlay that shows the accessibility information
+  /// reported by the framework.
+  final bool showSemanticsDebugger;
+
+  /// {@macro flutter.widgets.widgetsApp.debugShowCheckedModeBanner}
+  final bool debugShowCheckedModeBanner;
+
+  @override
+  _CupertinoAppState createState() => new _CupertinoAppState();
+}
+
+class _AlwaysCupertinoScrollBehavior extends ScrollBehavior {
+  @override
+  Widget buildViewportChrome(BuildContext context, Widget child, AxisDirection axisDirection) {
+    // Never build any overscroll glow indicators.
+    return child;
+  }
+
+  @override
+  ScrollPhysics getScrollPhysics(BuildContext context) {
+    return const BouncingScrollPhysics();
+  }
+}
+
+class _CupertinoAppState extends State<CupertinoApp> {
+  @override
+  void initState() {
+    super.initState();
+    _updateNavigator();
+  }
+
+  @override
+  void didUpdateWidget(CupertinoApp oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    _updateNavigator();
+  }
+
+  bool _haveNavigator;
+  void _updateNavigator() {
+    _haveNavigator = widget.home != null ||
+                     widget.routes.isNotEmpty ||
+                     widget.onGenerateRoute != null ||
+                     widget.onUnknownRoute != null;
+  }
+
+  // The `child` coming back out from WidgetsApp will always be null since
+  // we never passed in anything for it to create a Navigator inside
+  // WidgetsApp.
```

Comment:
add an assert to check that

Reply:
Done. Added some comments here and below too about not letting WidgetsApp make a navigator.